File: C++/231.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    bool isPowerOfTwo(int n) {
        std::int64_t c = 1;
        while (c <= n) {
            if (c == n)
                return true;
            c <<= 1;
        }
        return false;
    }
};
__________________________________________________________________________________________________
sample 7804 kb submission
class Solution {
public:
    bool isPowerOfTwo(int n) {
        return n > 0 && (n & (n - 1)) == 0;
    }
};
__________________________________________________________________________________________________

File: Java/231.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
	public boolean isPowerOfTwo(int n) {
		return (n > 0 && (n & (n - 1)) == 0);
	}
}
__________________________________________________________________________________________________
sample 32052 kb submission
class Solution {
    public boolean isPowerOfTwo(int n) {
        long i = 1;
        while(i < n){
            i *= 2;
        }
        
        return i == n;
        
    }
}
__________________________________________________________________________________________________
